# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.h
# Commit: 360b12c6b3e4
# Full Hash: 360b12c6b3e4e74203ba90324580a6c5e149b66b
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-07-16 04:07:00
# Regressor Bug: 1563604
# File Overlap Count: 1
# Description:
#   Bug 1563604 - Part 1: Confirm detach calls before dropping refs, r=peterv
#   
#   This ensures that the object is not discarded while there are still in-flight
#   messages from the parent process, and vice-versa, which could cause crashes if a
#   CC was run before any pending messages have arrived.
# ==============================================================================

diff -r 80fff55bee77 -r 360b12c6b3e4 dom/ipc/ContentChild.h
--- a/dom/ipc/ContentChild.h	Mon Jul 15 17:19:17 2019 +0000
+++ b/dom/ipc/ContentChild.h	Mon Jul 15 20:44:50 2019 +0000
@@ -717,7 +717,8 @@
   mozilla::ipc::IPCResult RecvAttachBrowsingContext(
       BrowsingContext::IPCInitializer&& aInit);
 
-  mozilla::ipc::IPCResult RecvDetachBrowsingContext(BrowsingContext* aContext);
+  mozilla::ipc::IPCResult RecvDetachBrowsingContext(
+      uint64_t aContextId, DetachBrowsingContextResolver&& aResolve);
 
   mozilla::ipc::IPCResult RecvCacheBrowsingContextChildren(
       BrowsingContext* aContext);